realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,210 @@
+            Assert.assertFalse(realm.schema.contains(newClass))
+            realm.cancelTransaction()
+            Assert.assertTrue(realm.schema.contains(existingClass))
+            Assert.assertFalse(realm.schema.contains(newClass))
+        }
+    }
+
+    @Test
+    fun disallow_removeField() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        val realm = DynamicRealm.getInstance(config!!)
+        val className = "StringOnly"
+        val fieldName = "chars"
+        val objectSchema = realm.schema[className]
+        Assert.assertNotNull(objectSchema)
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.beginTransaction()
+        assertFailsWith<IllegalArgumentException> {
+            // FIXME Why? We cannot update a sync schema "destructively"? -> Update Doc
+            //  Shouldn't this be UnsupportedOperation to be consistent with ex. addPrimaryKey, etc.
+            objectSchema.removeField(fieldName)
+        }
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.cancelTransaction()
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.close()
+    }
+
+    @Test
+    fun allow_addField() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        val className = "StringOnly"
+        val realm = DynamicRealm.getInstance(config!!)
+        val objectSchema = realm.schema[className]
+        Assert.assertNotNull(objectSchema)
+        realm.beginTransaction()
+        objectSchema!!.addField("foo", String::class.java)
+        Assert.assertTrue(objectSchema.hasField("foo"))
+        realm.commitTransaction()
+        Assert.assertTrue(objectSchema.hasField("foo"))
+        realm.close()
+    }
+
+    @Test
+    fun addPrimaryKey_notAllowed() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        val className = "StringOnly"
+        val fieldName = "chars"
+        val realm = DynamicRealm.getInstance(config!!)
+        val objectSchema = realm.schema[className]
+        Assert.assertNotNull(objectSchema)
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.beginTransaction()
+        assertFailsWith<java.lang.UnsupportedOperationException> {
+            objectSchema.addPrimaryKey(fieldName)
+        }
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.cancelTransaction()
+        Assert.assertTrue(objectSchema!!.hasField(fieldName))
+        realm.close()
+    }
+
+    @Test
+    fun addField_withPrimaryKeyModifier_notAllowed() {
+        // Init schema
+        Realm.getInstance(config!!).close()
+        val className = "StringOnly"
+
+        DynamicRealm.getInstance(config!!).use {realm ->
+            realm.beginTransaction()
+            val objectSchema = realm.schema[className]
+            Assert.assertNotNull(objectSchema)
+            // FIXME Is this in sync with expectations? Seems to be due to immutable scheme but test
```

Comment:
Yes, this is correct. Primary keys cannot be changed when using sync. They had good reasons, but I cannot remember them right now.